### PR TITLE
Feature/75462 webhook authentication rbac jwt token with FTs

### PR DESCRIPTION
**Pull request by prashantmgupta**
FTs for PBI 75462 has been added with Authentication Code generation logic.

### CODE REVIEW

**Review by RaviK2233 — APPROVED** [review]
Reviewed FT and everything looks good.

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.AutoTestClientIdNoRole, _config.testConfig.AzureadConfiguration.ClientSecretNoRole, AzureADToken);
+            }
+            else
+            {
+                AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.AutoTestClientId, _config.testConfig.AzureadConfiguration.ClientSecret, AzureADToken);
+            }
+            return AzureADToken;
+        }
+
+        private static async Task<string> GenerateAzureADToken(string ClientId, string ClientSecret, string Token)
```
Comment (jrippington) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:36]: Parameters should start lower-case, so 'clientId', 'clientSecret' etc

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        public async Task<string> GetAzureADToken()
+        {
+            _config = new();
+
+            AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.ClientId, _config.testConfig.AzureadConfiguration.ClientSecret, AzureADToken);
+            return AzureADToken;
+        }
+
+        public async Task<string> GetAzureADToken(Boolean noRole)
```
Comment (jrippington) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:21]: Use bool, not Boolean?

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Identity.Client;
+using Newtonsoft.Json.Linq;
+using UKHO.ERPFacade.API.FunctionalTests.FunctionalTests;
+
+namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
+{
+    public class ADAuthTokenProvider
+    {
+        static string AzureADToken = null;
+        public static Config _config;
```
Comment (jrippington) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:11]: This should be private scope, with an accessor, either a set accessor or an explicit SetConfig() method

**Review by jrippington — CHANGES_REQUESTED** [review]

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs (hunk 1 above)
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:36]: incorporated changes.

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs (hunk 2 above)
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:21]: incorporated changes.

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs (hunk 3 above)
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs:11]: incorporated changes.

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Pull request merged**